app/models/specific_tube_rack_creation.rb

```diff
@@ -0,0 +1,436 @@
+# frozen_string_literal: true
+# Allows for the creation of multiple tube racks and their tubes.
+# rubocop:disable Metrics/ClassLength
+class SpecificTubeRackCreation < AssetCreation
+  # Allows a many to many relationship between SpecificTubeRackCreations and child Tube racks.
+  class ChildTubeRack < ApplicationRecord
+    self.table_name = ('specific_tube_rack_creation_children')
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack
+  end
+
+  # Allows a many to many relationship between SpecificTubeRackCreations and Purposes.
+  class ChildPurpose < ApplicationRecord
+    self.table_name = 'specific_tube_rack_creation_purposes'
+    belongs_to :specific_tube_rack_creation
+    belongs_to :tube_rack_purpose, class_name: 'Purpose'
+  end
```

Comment:
Where's the link between rack and purpose? 
Should it be instead that `SpecificTubeRackCreation::ChildTubeRack` has a relationship `belongs_to :purpose`, or just a string with the purpose name or something?

Reply:
Link between rack and purpose is in the TubeRack model / resource?
This endpoint can make multiple tube racks of different purposes remember.
I copied the pattern from SpecificTubeCreation.
Can discuss.

Reply:
Looking at this again, I think you are right, there is a problem here.
It should be just the one table, not two.

Reply:
I have tried to change it for one table with belongs_to on purpose added but it breaks all the unit tests. I haven't been able to fix it.